File: picturemodel.cpp
#include "picturemodel.h"
#include <QDebug>

PictureModel::PictureModel(QObject *parent, QImage *image) : QAbstractTableModel(parent), image_(image)
{
}

int PictureModel::rowCount(const QModelIndex &parent) const
{
    if(image_ == 0){
        return 0;
    }
    return image_->height();
}

int PictureModel::columnCount(const QModelIndex &parent) const
{
    if(image_ == 0)
        return 0;
    return image_->width();
}


QVariant PictureModel::data(const QModelIndex &index, int role) const
{
    if (role == Qt::BackgroundRole){
            QRgb rgb = image_->pixel(index.column(), index.row());
            return QColor(rgb);
    }
    return QVariant();
}

void PictureModel::setImage(QImage *image)
{
    image_ = image;
    //for(int i = 0; i < rowCount(); ++i){
    //    qDebug() <<
    //}
    emit layoutChanged();
}

File: python/picture_model.py
from PyQt5.QtCore import QAbstractTableModel, QVariant
from PyQt5.QtGui import QColor


class PictureModel(QAbstractTableModel):
    def __init__(self, image=None, file=None):
        super.__init__()
        self.image = image
        self.file = file

    def rowCount(self, parent):
        if self.image is None:
            return 0
        return self.image.height()

    def columnCount(self, parent):
        if self.image is None:
            return 0
        return self.image.width()

    def data(self, index, role):
        if role == Qt.BackgroundRole:
            rgb = self.image.pixel(index.column(), index.row())
            return QColor(rgb)
        return QVariant()

    def open_image(self, file_name):
        pass

    def save_image(self, file_name=None):
        pass
